conans/client/api/conan_api.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import os
+import sys
+import time
+from io import StringIO
+
+from conans import __version__ as client_version
+from conans.client.cache.cache import ClientCache
+from conans.client.graph.graph_binaries import GraphBinariesAnalyzer
+from conans.client.graph.graph_manager import GraphManager
+from conans.client.graph.proxy import ConanProxy
+from conans.client.graph.python_requires import ConanPythonRequire, PyRequireLoader
+from conans.client.graph.range_resolver import RangeResolver
+from conans.client.hook_manager import HookManager
+from conans.client.loader import ConanFileLoader
+from conans.client.migrations import ClientMigrator
+from conans.cli.output import ConanOutput
+from conans.client.remote_manager import RemoteManager
+from conans.client.rest.auth_manager import ConanApiAuthManager
+from conans.client.rest.conan_requester import ConanRequester
+from conans.client.rest.rest_client import RestApiClientFactory
+from conans.client.runner import ConanRunner
+from conans.client.store.localdb import LocalDB
+from conans.client.tools.env import environment_append
+from conans.client.userio import UserIO
+from conans.errors import NoRemoteAvailable
+from conans.model.version import Version
+from conans.paths import get_conan_user_home
+from conans.tools import set_global_instances
+from conans.util.conan_v2_mode import CONAN_V2_MODE_ENVVAR
+from conans.util.env_reader import get_env
+from conans.util.files import exception_message_safe
+from conans.util.log import configure_logger
+from conans.util.tracer import log_command, log_exception
+from tqdm import tqdm
+
+
+def api_method(f):
+    def wrapper(api, *args, **kwargs):
+        try:  # getcwd can fail if Conan runs on an unexisting folder
+            old_curdir = os.getcwd()
+        except EnvironmentError:
+            old_curdir = None
+        try:
+            api.create_app()
+            log_command(f.__name__, kwargs)
+            with environment_append(api.app.cache.config.env_vars):
+                return f(api, *args, **kwargs)
+        except Exception as exc:
+            msg = exception_message_safe(exc)
+            try:
+                log_exception(exc, msg)
+            except BaseException:
+                pass
+            raise
+        finally:
+            if old_curdir:
+                os.chdir(old_curdir)
+
+    return wrapper
+
+
+class ConanApp(object):
```

Comment:
If this is not really used at this moment, better simplify and leave it empty?

Reply:
It's used here: https://github.com/conan-io/conan/blob/6bc8d4593aff1585e387ebece3b849e102f806ad/conans/client/api/conan_api.py#L47 
But I'll get rid of it and maybe it could be the start of rethinking the relationship between the API, ConanApp and collaborators...